test/server/cards/01_SOR/events/Restock.spec.ts

```diff
@@ -0,0 +1,67 @@
+describe('Restock', function() {
+    integration(function(contextRef) {
+        describe('Restock\'s ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['restock'],
+                        discard: ['pyke-sentinel', 'atst', 'battlefield-marine', 'resupply'],
+                    },
+                    player2: {
+                        discard: ['consular-security-force', 'specforce-soldier', 'echo-base-defender']
+                    }
+                });
+            });
+
+            it('should return up to 4 card from our discard to bottom of deck', function () {
+                const { context } = contextRef;
+
+                // play restock
+                context.player1.clickCard(context.restock);
+
+                expect(context.player1).toHaveExactPromptButtons(['Your discard', 'Opponent discard']);
+                context.player1.clickPrompt('Your discard');
+
+                expect(context.player1).toBeAbleToSelectExactly([context.pykeSentinel, context.atst, context.battlefieldMarine, context.resupply, context.restock]);
+
+                // select cards
+                context.player1.clickCard(context.resupply);
+                context.player1.clickCard(context.restock);
+                context.player1.clickCard(context.battlefieldMarine);
+                context.player1.clickCard(context.atst);
```

Comment:
We should confirm the 4 card restriction with `clickCardNonChecking()` and confirm that the fifth clicked card stays in the deck